mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+	opts := options.ChangeStream().SetMaxAwaitTime(2 * time.Second)
+	changeStream, err := client.Watch(context.TODO(), mongo.Pipeline{matchStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// print out all change stream events in the order they're received
+	// see the mongo.ChangeStream documentation for more examples of using change streams
+	for changeStream.Next(context.TODO()) {
+		fmt.Println(changeStream.Current)
+	}
+}
+
+// Database examples
+
+func ExampleDatabase_Aggregate() {
+	var db *mongo.Database
+
+	// specify a pipeline that will list all sessions on the target server
+	// specify the MaxTime option to limit the amount of time the operation can run on the server
+	localSessionsStage := bson.D{{"$listLocalSessions", bson.D{{"allUsers", true}}}}
+	opts := options.Aggregate().SetMaxTime(2 * time.Second)
+	cursor, err := db.Aggregate(context.TODO(), mongo.Pipeline{localSessionsStage}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned documents and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.M
+	if err = cursor.All(context.TODO(), &results); err != nil {
+		log.Fatal(err)
+	}
+	for _, result := range results {
+		fmt.Println(result)
+	}
+}
+
+func ExampleDatabase_ListCollectionNames() {
+	var db *mongo.Database
+
+	// use a filter to only select capped collections
+	result, err := db.ListCollectionNames(context.TODO(), bson.D{{"options.capped", true}})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, coll := range result {
+		fmt.Println(coll)
+	}
+}
+
+func ExampleDatabase_ListCollections() {
+	var db *mongo.Database
+
+	// use a filter to only select capped collections
+	cursor, err := db.ListCollections(context.TODO(), bson.D{{"options.capped", "true"}})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// get a list of all returned specifications and print them out
+	// see the mongo.Cursor documentation for more examples of using cursors
+	var results []bson.M
+	if err = cursor.All(context.TODO(), &results); err != nil {
+		log.Fatal(err)
+	}
+	for _, result := range results {
+		fmt.Println(result)
+	}
+}
+
+func ExampleDatabase_RunCommand() {
+	var db *mongo.Database
+
+	// create a capped collection named "capped-coll"
+	// specify the ReadPreference option to explicitly set the read preference to primary
+	command := bson.D{{"create", "capped-coll"}, {"capped", true}, {"size", 4096}}
+	opts := options.RunCmd().SetReadPreference(readpref.Primary())
+	var result bson.M
+	if err := db.RunCommand(context.TODO(), command, opts).Decode(&result); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(result)
+}
+
+func ExampleDatabase_RunCommandCursor() {
+	var db *mongo.Database
+
+	// run a find command against collection "foo"
+	// specify the ReadPreference option to explicitly set the read preference to primary
+	command := bson.D{{"find", "foo"}}
+	opts := options.RunCmd().SetReadPreference(readpref.Primary())
```

Comment:
This example could confuse people, and make them wonder why not just use Find method (which they should).  Is there a more realistic use case?  If not, best to just remove this example entirely.

Reply:
I couldn't think of a more realistic one. IIRC this method was a request from tools (?) and is used to do things like find with UUID, which isn't really a common use case. Removed this example.